include/circt/Dialect/SystemC/SystemCTypes.td

```diff
@@ -18,12 +18,17 @@ include "circt/Dialect/SystemC/SystemCDialect.td"
 class SystemCType<Pred condition, string description, string cppClassName>
   : DialectType<SystemCDialect, condition, description, cppClassName>;
 
-// A handle to refer to circt::systemc::FuncHandleType in ODS.
-def FuncHandleType : SystemCType<
-    CPred<"::circt::hw::type_isa<circt::systemc::FuncHandleType>($_self)">,
-    "FuncHandleType",
-    "::circt::hw::TypeAliasOr<circt::systemc::FuncHandleType>">,
-  BuildableType<"::circt::systemc::FuncHandleType::get($_builder.getContext())">;
+// A handle to refer to a mlir::FunctionType with zero inputs and results,
+// i.e., a void function.
+def VoidFunctionType : SystemCType<
```

Comment:
Nit: this is a nullary void function. I don't know what the proper name is of a function with no results should be other than void.

Reply:
Good point. There does not seem to exist terminology for the "arity" of function results in mathematics. Thus I used the C language terminology for functions that do not return a value. I'll rename it to `NullaryVoidFunctionType` then.